### PR TITLE
[Feature] Support DIOR Dataset

### PR DESCRIPTION
**Pull request by yxzhao2022**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support DIOR Dataset.
Complete task1 in #626 

## Modification

 Implement the DIOR dataset and related documents.

 Performance:
| model             | backbone | map   | angle | lr schd |
|:-------------------:|:----------:|:------:|:------:|:--------:|
| rotated retinanet | resnet50 | 50.26 | oc    | 1x      |
| rotated retinanet | resnet50 | 53.94 | le90  | 1x      |

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tests/test_datasets/test_dior.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import unittest
+
+from mmrotate.datasets import DIORDataset
+
+
+class TestHRSCDataset(unittest.TestCase):
```
Comment (liuyanyi) [inline tests/test_datasets/test_dior.py:7]: TestDIORDataset

File: configs/_base_/datasets/dior.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                filter_cfg=dict(filter_empty_gt=True),
+                pipeline=train_pipeline),
+            dict(
+                type=dataset_type,
+                data_root=data_root,
+                ann_file='ImageSets/Main/val.txt',
+                data_prefix=dict(img_path='JPEGImages-trainval'),
+                filter_cfg=dict(filter_empty_gt=True),
+                pipeline=train_pipeline)
+        ]))
```
Comment (liuyanyi) [inline configs/_base_/datasets/dior.py:60]: I think the `ConcatDataset` could be simplified to 
```
dataset=dict(
    type=dataset_type,
    data_root=data_root,
    ann_file=['ImageSets/Main/train.txt', 'ImageSets/Main/val.txt'],
    data_prefix=dict(img_path='JPEGImages-trainval'),
    filter_cfg=dict(filter_empty_gt=True),
    pipeline=train_pipeline),
```

File: mmrotate/datasets/dior.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                    (0, 0, 192), (250, 170, 30), (100, 170, 30), (220, 220, 0),
+                    (175, 116, 175), (250, 0, 30), (165, 42, 42),
+                    (255, 77, 255), (0, 226, 252), (182, 182, 255), (0, 82, 0),
+                    (120, 166, 157)]
+    }
+
+    def __init__(self,
+                 ann_subdir: str = 'Annotations/Oriented Bounding Boxes/',
+                 file_client_args: dict = dict(backend='disk'),
+                 xmltype: str = 'obb',
```
Comment (liuyanyi) [inline mmrotate/datasets/dior.py:47]: `xmltype` may not well understood, maybe `ann_type` or other name?

File: tests/test_datasets/test_dior.py (hunk 1 above)
Comment (yxzhao2022) [reply tests/test_datasets/test_dior.py]: Sorry, I have updated.

File: mmrotate/datasets/dior.py (hunk 3 above)
Comment (yxzhao2022) [reply mmrotate/datasets/dior.py]: Thanks for your advice. I have updated.

File: configs/_base_/datasets/dior.py (hunk 2 above)
Comment (yxzhao2022) [reply configs/_base_/datasets/dior.py:60]: I've tried this before, but  it no longer seems to be supported in mmengine.

Comment (yxzhao2022) [reply configs/_base_/datasets/dior.py:60]: > I think the `ConcatDataset` could be simplified to
> 
> ```
> dataset=dict(
>     type=dataset_type,
>     data_root=data_root,
>     ann_file=['ImageSets/Main/train.txt', 'ImageSets/Main/val.txt'],
>     data_prefix=dict(img_path='JPEGImages-trainval'),
>     filter_cfg=dict(filter_empty_gt=True),
>     pipeline=train_pipeline),
> ```

I've tried this before, but it doesn't seem to be supported in `mmengine`.

**Review by zytx121 — APPROVED** [review]
LGTM

**Review by liuyanyi — APPROVED** [review]
LGTM

File: configs/rotated_retinanet/rotated-retinanet-rbox-oc_r50_fpn_1x_dior.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+_base_ = 'rotated-retinanet-rbox-le90_r50_fpn_1x_dior.py'
+
+angle_version = 'oc'
+
+model = dict(
```
Comment (RangiLyu) [inline configs/rotated_retinanet/rotated-retinanet-rbox-oc_r50_fpn_1x_dior.py:5]: Do we need to update the readme and upload pre-trained models?

Comment (yxzhao2022) [reply configs/rotated_retinanet/rotated-retinanet-rbox-oc_r50_fpn_1x_dior.py:5]: > Do we need to update the readme and upload pre-trained models?
Yes, I will update the readme and upload these models soon.

**Review by zytx121 — APPROVED** [review]

**Review by RangiLyu — APPROVED** [review]

**Pull request merged**